src/reynir/bintokenizer.py

```diff
@@ -1204,7 +1221,7 @@ def unknown_surname(tok: Tok) -> bool:
                 """ Check for unknown (non-Icelandic) surnames """
                 # Accept (most) upper case words as a surnames
                 if auto_uppercase:
-                    if tok.txt in UPPER_CASE_ABBREVS:
+                    if tok.txt in MIDDLE_NAME_ABBREVS:
```

Comment:
Might be good to update the comment below to accurately describe what's happening, because it's starting to get really complicated. In other words, let's have mercy on future programmers who will be trying to understand this code ;-)

Reply:
I realized this part of the function wasn't needed so I simplified it. Now it only deals with foreign middle names (unknown words) so it's a bit more straight-forward.